File: ViewPager/app/src/main/java/github/nisrulz/viewpager/MyFragmentPagerAdapter.java
package github.nisrulz.viewpager;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

/**
 * @author Nishant Srivastava
 */

public class MyFragmentPagerAdapter extends FragmentPagerAdapter {

    public MyFragmentPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public int getCount() {
        return 3;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return Page.newInstance(0, "Zeroth : Page Type 1");
            case 1:
                return Page2.newInstance(1, "One : Page Type 2");
            case 2:
                return Page.newInstance(2, "Two : Page Type 1");
            default:
                return Page2.newInstance(3, "Default : Page Type 2");
        }

    }
}

File: ViewPager/app/src/main/java/github/nisrulz/example/viewpager/MyFragmentPagerAdapter.kt
package github.nisrulz.example.viewpager

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class MyFragmentPagerAdapter(fm: FragmentManager) :
    FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    override fun getCount(): Int {
        return 3
    }

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> Page.newInstance(0, "Page Type 1")
            1 -> Page2.newInstance(1, "Page Type 2")
            2 -> Page.newInstance(2, "Page Type 1")
            else -> Page2.newInstance(3, "Page Type 2")
        }
    }
}